<launch>

  <node pkg="project4_extra" type="control_node.py" name="controller" output="screen">
    <param name="linear_spead" value="0.1"/>
    <param name="angular_k_p" value="2.75"/>
    <param name="angular_k_d" value="27.5"/>
    <param name="angular_k_i" value="0"/>
    <param name="dt" value="0.005"/>
    <param name="distance" value="0.3"/>
    <param name="epsilon" value="0.07"/>
    <param name="turn_direction" value="left"/>
  </node>

  <node type="laser_scan_assembler" pkg="laser_assembler" name="assembler">
    <param name="max_scans" type="int" value="4000"/>
    <param name="fixed_frame" type="string" value="odom"/>
  </node>

  <node pkg="project4_extra" type="laser_scaner_assembler_node.py" name="laser_assembler_pcd"/>

  <node pkg="project4_extra" type="monitor.py" name="monitor" ></node>

  <include file="$(find turtlebot3_gazebo)/launch/maze.launch">
    <arg name="x_pos" value="-0.5"/>
    <arg name="y_pos" value="0.0"/>
    <arg name="z_pos" value="0.0"/>
    <arg name="yaw" value="0.0"/>

  </include>  

  <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_gazebo_rviz.launch">
  </include>
</launch>
